h2o-genmodel/src/main/java/hex/genmodel/algos/deepwater/caffe/DeepwaterCaffeModel.java

```diff
@@ -174,53 +156,11 @@ public void train(float[] data, float[] label) {
 
   //
 
-  private void startDocker(String image, boolean gpu) throws IOException {
-    String home = System.getProperty("user.home");
-    int uid = Integer.parseInt(new BufferedReader(new InputStreamReader(
-        Runtime.getRuntime().exec("id -u").getInputStream())).readLine());
-    int gid = Integer.parseInt(new BufferedReader(new InputStreamReader(
-        Runtime.getRuntime().exec("id -g").getInputStream())).readLine());
-    String pwd = System.getProperty("user.dir") + "/caffe";
-
-    /*
-    // nuke all existing docker images! CAREFUL
-    ProcessBuilder pb = new ProcessBuilder("bash", "-c", "'docker stop $(docker ps -a -q)'");
-    pb.redirectError(ProcessBuilder.Redirect.INHERIT);
-    _process = pb.start();
-    try {
-      _process.waitFor();
-    } catch (InterruptedException e) { // Ignore
-    }
-    */
-
-    // Update image first
-    String s = "docker pull " + image;
-    ProcessBuilder pb = new ProcessBuilder(s.split(" "));
-    pb.redirectError(ProcessBuilder.Redirect.INHERIT);
-    _process = pb.start();
-    try {
-      _process.waitFor();
-    } catch (InterruptedException e) { // Ignore
-    }
-
-    // Launch it
-    String opts = "-i --rm --user " + uid + ":" + gid + " -v " + pwd + ":" + pwd + " -w " + pwd;
-    opts += " -v " + home + "/h2o-docker/caffe:/h2o-docker/caffe";
-    String tmp = System.getProperty("java.io.tmpdir");
-    opts += " -v " + tmp + ":" + tmp;
-    s = gpu ? "nvidia-docker" : "docker";
-    s += " run " + opts + " " + image + " python3 /h2o-docker/caffe/backend.py";
-    pb = new ProcessBuilder(s.split(" "));
-    pb.redirectError(ProcessBuilder.Redirect.INHERIT);
-    _process = pb.start();
-  }
-
   // Debug, or if wee find a way to package Caffe without Docker
   private void startRegular() throws IOException {
-    String home = System.getProperty("user.home");
-    String pwd = home + "/h2o-docker/caffe";
+    String pwd = "/opt/caffe-h2o";
     ProcessBuilder pb = new ProcessBuilder("python3 backend.py".split(" "));
-    pb.environment().put("PYTHONPATH", home + "/caffe/python:" + home + "/protobuf/python");
+    pb.environment().put("PYTHONPATH", "/opt/caffe/python");
```

Comment:
Caffe only runs in the docker image and we want it always to be in that one place. If it's not installed there we don't consider caffe to have been installed and it will not be instantiated.